### DISCUSSION TITLE
Auto face detailing step

### DISCUSSION THREAD

**Discussion by patrickchamp** [lllyasviel/Fooocus#2467] [Ideas]
When generating images, it’s almost guaranteed that I’ll need to improve face detail. It would be great to have fooocus do a simple automated face restore or adetailer extension-type face detection & detailing as a post processing step.

This would simplify the workflow and cut down on the need to use the input image feature entirely for many generations.

**Comment by Amit30swgoh** [comment]
If I split the work up into four tiles, each part would have 1024x1024 pixels and then merged them into one 4K image, most of the inaccuracies come from the lack of pixels since most SDXL models were trained on less than 2k photos around 1024x1024 pixels.

We need five steps to improve the cutting part in the horizon and vertical line where we splitted the image so it will look like one image if the script auto-cuts the image into four and uploads each part and improves the detail. 

I've thought about it a lot, but I don't know how to code it, so if someone does it, it will be amazing, and can change the way we use SDXL in the future.

Let's wait and see who wins, it's just a matter of time and effort.

If I didn't think it was possible if you were a great developer, I wouldn't think about it much...
